connector/servicegraphconnector/connector.go

```diff
@@ -215,6 +216,8 @@ func (p *serviceGraphConnector) metricFlushLoop(flushInterval time.Duration) {
 }
 
 func (p *serviceGraphConnector) flushMetrics(ctx context.Context) error {
+	p.store.Expire()
```

Comment:
Edges are already expired in a different goroutine, why force expiration here?

Reply:
That's what I saw, but without it (or because it's a different goroutine), the shutdown and metrics collection will happen before the edge expire&complete, so metrics from pending edges are not collected (they won't be added to `.reqTotal` for example).

Do you know if there's a better way to test this?

PS: This is unrelated to the addition of the new label, I realized when I was adding a test for `connection_type: virtual_node`.

Reply:
OK fixed, I overlook a minor detail about the metric flush loop, so test has to allow it. No more double-expire here.